Compiler error when schema contains a property named "system" and a dateTime

DateTime properties cause the following code snippet to get generated:
```cs
            /// <summary>
            /// Initializes a new instance of the <see cref="EventStartDatetimeEntity"/> struct.
            /// </summary>
            /// <param name="value">The value from which to construct the instance.</param>
            public EventStartDatetimeEntity(in ReadOnlySpan<byte> value)
            {
                this.backing = Backing.String;
                this.jsonElementBacking = default;

#if NET8_0_OR_GREATER
                this.stringBacking = System.Text.Encoding.UTF8.GetString(value);
#else
                byte[] bytes = ArrayPool<byte>.Shared.Rent(value.Length);
                try
                {
                    value.CopyTo(bytes);
                    this.stringBacking = System.Text.Encoding.UTF8.GetString(bytes);
                }
                finally
                {
                    ArrayPool<byte>.Shared.Return(bytes);
                }
#endif
            }
```


If you have a property called `System` in a parent object, this causes the following error:
```
/home/fernie/scratch/repro/obj/Debug/net8.0/generated/Corvus.Json.SourceGenerator/Corvus.Json.SourceGenerator.IncrementalSourceGenerator/SchemaObject.SystemEntity.EventStartDatetimeEntity.String_g.cs(62,38): error CS0120: An object reference is required for the non-static field, method, or property 'SchemaObject.System' [/home/fernie/scratch/repro/repro.csproj]
/home/fernie/scratch/repro/obj/Debug/net8.0/generated/Corvus.Json.SourceGenerator/Corvus.Json.SourceGenerator.IncrementalSourceGenerator/SchemaObject.SystemEntity.EventStartDatetimeEntity.String_g.cs(369,40): error CS0120: An object reference is required for the non-static field, method, or property 'SchemaObject.System' [/home/fernie/scratch/repro/repro.csproj]
/home/fernie/scratch/repro/obj/Debug/net8.0/generated/Corvus.Json.SourceGenerator/Corvus.Json.SourceGenerator.IncrementalSourceGenerator/SchemaObject.SystemEntity.EventStartDatetimeEntity.String_g.cs(379,39): error CS0120: An object reference is required for the non-static field, method, or property 'SchemaObject.System' [/home/fernie/scratch/repro/repro.csproj]
```

because it assumes the `System`  in `System.Text.Encoding.UTF8.GetString(value)` is referring to the new property.

Perhaps it needs to be prefixed with a `global::`?

I've also attached a minimal repro case with a schema and project:
[repro.zip](https://github.com/user-attachments/files/20375619/repro.zip)

--- a/Solutions/Corvus.Json.CodeGeneration.CSharp/Corvus.Json.CodeGeneration/CSharp/Formatting.cs
+++ b/Solutions/Corvus.Json.CodeGeneration.CSharp/Corvus.Json.CodeGeneration/CSharp/Formatting.cs
@@ -106,6 +106,7 @@ public static class Formatting
         "SchemaLocation",
         "SetProperty",
         "Stream",
+        "System",
         "ToString",
         "TryCreateUriTemplateParser",
         "TryGetBoolean",

--- a/Solutions/Corvus.Json.Specs/Features/AdditionalSchema/Draft202012/collisionWithBuiltInName.feature
+++ b/Solutions/Corvus.Json.Specs/Features/AdditionalSchema/Draft202012/collisionWithBuiltInName.feature
@@ -2,7 +2,7 @@
 
 Feature: Collision with a built-in name draft2020-12
 
-Scenario Outline: An property that matches a built-in name
+Scenario Outline: A property that matches a built-in name
 	Given a schema file
 		"""
 		{
@@ -25,3 +25,33 @@ Examples:
 	| inputData         | valid |
 	| {"match": "foo" } | true  |
 	| {"match": "f" }   | false |
+
+
+Scenario Outline: Repro 625 - collides with System
+	Given a schema file
+		"""
+		{
+		  "type": "object",
+		  "properties": {
+			"system": {
+			  "type": "string",
+			  "minLength": 0
+			},
+			"someOtherProp": {
+			  "type": "string",
+			  "minLength": 0
+			}
+		  },
+		  "required": ["system", "someOtherProp"]
+		}
+		"""
+	And the input data value <inputData>
+	And I generate a type for the schema
+	And I construct an instance of the schema type from the data
+	When I validate the instance
+	Then the result will be <valid>
+
+Examples:
+	| inputData                                  | valid |
+	| {"system": "foo", "someOtherProp": "bar" } | true  |
+	| {"someOtherProp": "bar"  }                 | false |

